clarity/src/vm/functions/principals.rs

```diff
@@ -38,13 +69,259 @@ pub fn special_is_standard(
         _ => return Err(CheckErrors::TypeValueError(TypeSignature::PrincipalType, owner).into()),
     };
 
-    let address_is_mainnet = version == C32_ADDRESS_VERSION_MAINNET_MULTISIG
-        || version == C32_ADDRESS_VERSION_MAINNET_SINGLESIG;
-    let address_is_testnet = version == C32_ADDRESS_VERSION_TESTNET_MULTISIG
-        || version == C32_ADDRESS_VERSION_TESTNET_SINGLESIG;
-    let context_is_mainnet = env.global_context.mainnet;
+    Ok(Value::Bool(version_matches_current_network(
+        version,
+        env.global_context,
+    )))
+}
+
+/// Creates a Tuple which is the result of parsing a Principal tuple into a Tuple of its `version`
+/// and `hash-bytes`.
+fn create_principal_parse_tuple(
+    version: u8,
+    hash_bytes: &[u8; 20],
+    name_opt: Option<ContractName>,
+) -> Value {
+    Value::Tuple(
+        TupleData::from_data(vec![
+            (
+                "version".into(),
+                Value::Sequence(SequenceData::Buffer(BuffData {
+                    data: vec![version],
+                })),
+            ),
+            (
+                "hash-bytes".into(),
+                Value::Sequence(SequenceData::Buffer(BuffData {
+                    data: hash_bytes.to_vec(),
+                })),
+            ),
+            (
+                "name".into(),
+                Value::Optional(OptionalData {
+                    data: name_opt.map(|name| Box::new(Value::from(name))),
+                }),
+            ),
+        ])
+        .expect("FAIL: Failed to initialize tuple."),
+    )
+}
+
+/// Creates Response return type, to wrap an *actual error* result of a `principal-construct` or
+/// `principal-parse`.
+///
+/// The response is an error Response, where the `err` value is a tuple `{error_int,parse_tuple}`.
+/// `error_int` is of type `UInt`, `parse_tuple` is None.
+fn create_principal_true_error_response(error_int: u32) -> Value {
+    Value::Response(ResponseData {
+        committed: false,
+        data: Box::new(Value::Tuple(
+            TupleData::from_data(vec![
+                ("error_int".into(), Value::UInt(error_int.into())),
+                ("value".into(), Value::none()),
+            ])
+            .expect("FAIL: Failed to initialize tuple."),
+        )),
+    })
+}
+
+/// Creates Response return type, to wrap a *return value returned as an error* result of a
+/// `principal-construct` or `principal-parse`.
```

Comment:
this function seems to be used for `principal-construct` - same thing for `create_principal_true_error_response`

Reply:
Fixed